nvtabular/workflow/workflow.py

```diff
@@ -96,7 +98,15 @@ def transform(self, dataset: Dataset) -> Dataset:
         Dataset
             Transformed Dataset with the workflow graph applied to it
         """
-        return self._transform_impl(dataset)
+        if isinstance(dataset, Dataset):
+            return self._transform_impl(dataset)
+        elif isinstance(dataset, pd.DataFrame) or (cudf and isinstance(dataset, cudf.DataFrame)):
+            return self._transform_df(dataset)
+        else:
+            raise ValueError(
+                "Workflow.transform received an unsupported type: {type(dataset)} "
+                "Supported types are a `merlin.io.Dataset` or DataFrame (pandas or cudf)"
+            )
```

Comment:
I also wonder whether this might be a use case for either `singledispatch` or `lazy_singledispatch` since this code is essentially doing type-based dispatch anyway.

Reply:
`singledispatchmethod` seems to  work relatively well here